scripts/download_file.py

```diff
@@ -0,0 +1,156 @@
+
+import argparse
+import urllib.request
+import urllib.error
+import os
+import hashlib
+import sys
+
+
+parser = argparse.ArgumentParser(description="Download data file for EFI KBase app")
+parser.add_argument("--remote-dir", required=True, type=str, help="URL of the directory in which the file resides")
+parser.add_argument("--remote-file", required=True, type=str, help="Name of the file on the remote to download")
+parser.add_argument("--local-dir", required=True, type=str, help="Path to local directory where file will be unpacked")
+parser.add_argument("--local-file", required=True, type=str, help="Path to file on local system where download file is placed")
+args = parser.parse_args()
+
+
+if not os.path.isdir(args.local_dir):
+    os.mkdir(args.local_dir)
+
+
+
+
+
+
+def get_remote_file_contents(remote_file):
+    """
+    Downloads the file as a string and return to user; if HTTP error then returns False
+    """
+    try:
+        response = urllib.request.urlopen(remote_file)
+    except urllib.error.HTTPError as e:
+        print(f"Unable to urlopen {remote_file}: {e}")
+        return False
+    file_contents = response.read()
+    file_contents = str(file_contents.decode('utf-8'))
+    return file_contents
+
+
+def get_file_list(args):
+    """
+    Retrieves a list of files on the remote that need to be fetched
+    """
+    remote_file = args.remote_dir + "/" + args.remote_file + ".file_list"
```

Comment:
Consider using `urllib.parse.urljoin`

Reply:
I looked into pathlib; it's for file system paths not URLs.  I elected to keep using string concatenation here because the application is very specific.  URL separators are forward slashes (`/`) so I think doing the concatenation here is acceptable.

Reply:
Linux paths use forward slashes, so pathlib works as expected in my working environment. I am happily naive about how Windows formats paths.